scripts/cluster-install/install.py

```diff
@@ -0,0 +1,392 @@
+from __future__ import print_function
+import os
+import yaml
+import time
+import string
+import random
+import logging
+import paramiko
+import argparse
+from openstack_connection import create_openstack_vms, handle_failure, delete_vm
+
+CERT_PATH = '~/.cloudify-test-ca'
+LOCAL_INSTALL_CLUSTER = '/tmp/install_cluster'
+LOCAL_INSTALL_CLUSTER_CERTS = '{}/certs'.format(LOCAL_INSTALL_CLUSTER)
+REMOTE_PARENT_DIRECTORY = '/tmp'
+REMOTE_INSTALL_CLUSTER = '{}/install_cluster'.format(REMOTE_PARENT_DIRECTORY)
+
+
+def retry_with_sleep(func, *func_args, **kwargs):
+    retry_count = kwargs.get('retry_count', 10)
+    delay = kwargs.get('delay', 2)
+    for i in range(retry_count):
+        try:
+            return func(*func_args)
+        except Exception as e:
+            if i < retry_count - 1:
+                time.sleep(delay)
+                continue
+            else:
+                raise e
+
+
+def _create_ssh_client_func(hostname, username, key_file):
+    client = paramiko.SSHClient()
+    client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+    client.connect(hostname=hostname, username=username, key_filename=key_file)
+    return client
+
+
+def _create_ssh_client(hostname, username, key_file):
```

Comment:
ideally close the client when the program is done

Reply:
Excellent point, I remember you also pointed that out during the demo as the reason for the exception I got at the end. Fixed it now.